internal/handlers/common/aggregations/stages/group.go

```diff
@@ -165,73 +171,102 @@ func (g *group) Process(ctx context.Context, iter types.DocumentsIterator, close
 
 // groupDocuments groups documents by group expression.
 func (g *group) groupDocuments(ctx context.Context, in []*types.Document) ([]groupedDocuments, error) {
-	groupKey, ok := g.groupExpression.(string)
-	if !ok {
-		// non-string key aggregates values of all `in` documents into one aggregated document.
-		return []groupedDocuments{{
-			groupID:   g.groupExpression,
-			documents: in,
-		}}, nil
-	}
+	switch groupKey := g.groupExpression.(type) {
+	case *types.Document:
+		if !operators.IsOperator(groupKey) {
+			break
+		}
 
-	expression, err := aggregations.NewExpression(groupKey)
-	if err != nil {
-		var exprErr *aggregations.ExpressionError
-		if !errors.As(err, &exprErr) {
-			return nil, lazyerrors.Error(err)
+		op, err := operators.NewOperator(groupKey)
+		if err != nil {
+			return nil, err
 		}
 
-		switch exprErr.Code() {
-		case aggregations.ErrNotExpression:
-			// constant value aggregates values of all `in` documents into one aggregated document.
-			return []groupedDocuments{{
-				groupID:   groupKey,
-				documents: in,
-			}}, nil
-		case aggregations.ErrEmptyFieldPath:
-			return nil, commonerrors.NewCommandErrorMsgWithArgument(
-				// TODO
-				commonerrors.ErrGroupInvalidFieldPath,
-				"'$' by itself is not a valid Expression",
-				"$group (stage)",
-			)
-		case aggregations.ErrInvalidExpression:
-			return nil, commonerrors.NewCommandErrorMsgWithArgument(
-				commonerrors.ErrFailedToParse,
-				fmt.Sprintf("'%s' starts with an invalid character for a user variable name", types.FormatAnyValue(groupKey)),
-				"$group (stage)",
-			)
-		case aggregations.ErrEmptyVariable:
-			return nil, commonerrors.NewCommandErrorMsgWithArgument(
-				commonerrors.ErrFailedToParse,
-				"empty variable names are not allowed",
-				"$group (stage)",
-			)
-		// TODO https://github.com/FerretDB/FerretDB/issues/2275
-		case aggregations.ErrUndefinedVariable:
-			return nil, commonerrors.NewCommandErrorMsgWithArgument(
-				commonerrors.ErrGroupUndefinedVariable,
-				fmt.Sprintf("Use of undefined variable: %s", types.FormatAnyValue(groupKey)),
-				"$group (stage)",
-			)
-		default:
-			panic(fmt.Sprintf("unhandled field path error %s", exprErr.Error()))
+		var group groupMap
+
+		for _, doc := range in {
+			val, err := op.Process(doc)
+			if err != nil {
+				return nil, err
```

Comment:
Also we want to cover error case by test. When I was testing this with $sum with projection, I used recursive $sum to error like below
https://github.com/FerretDB/FerretDB/blob/4d534d602e82d7726f327c2915d73cc253195553/integration/aggregate_documents_compat_test.go#L1924